test/utils.jl

```diff
@@ -0,0 +1,69 @@
+@testset "utils.jl" begin
+    @testset "submat" begin
+
+        @testset "PDMat" begin
+            M = PDMat(TEST_MATRICES["Positive definite"])
+            M_dense = Matrix(M)
+
+            subM = submat(M, [1, 3])
+            @test subM isa PDMat
+            @test Matrix(subM) == M_dense[[1, 3], [1, 3]]
+        end
+
+        @testset "PDDiagMat" begin
+            M = PDiagMat(diag(TEST_MATRICES["Positive definite"]))
+            M_dense = Matrix(M)
+
+            subM = submat(M, [1, 3])
+            @test subM isa PDiagMat
+            @test Matrix(subM) == M_dense[[1, 3], [1, 3]]
+        end
+
+        @testset "PDSparseMat" begin
+            M = PDSparseMat(sparse(TEST_MATRICES["Positive definite"]))
+            M_dense = Matrix(M)
+
+            subM = submat(M, [1, 3])
+            @test subM isa PDSparseMat
+            @test Matrix(subM) == M_dense[[1, 3], [1, 3]]
+        end
+
+        @testset "ScalMat" begin
+            M = ScalMat(3, 0.1)
+            M_dense = Matrix(M)
+
+            subM = submat(M, [1, 3])
+            @test subM isa ScalMat
+            @test Matrix(subM) == M_dense[[1, 3], [1, 3]]
+            @test_throws BoundsError submat(M, [1, 10])
+        end
+
+        @testset "PSDMat" begin
+            SM = TEST_MATRICES["Positive semi-definite"]
+            pivoted = cholesky(SM, Val(true); check=false)
+            M = PSDMat(SM, pivoted)
```

Comment:
just curious, is this the way how we should construct a PSDMat, i.e. provide the `cholesky` in addition to the matrix, and have to set `check` as false?

Reply:
I don't think you have to provide the cholesky explicity - I was just following the example of the existing tests. But the docs would suggest I could have just passed `SM` by itself https://github.com/invenia/PDMatsExtras.jl#psdmat